src/main/java/BuyAndSellStock2.java

```diff
@@ -0,0 +1,40 @@
+package main.java;
+
+public class BuyAndSellStock2 {
+
+    public int maxProfit(int[] prices) {
+
+        // Assume invalid inputs return a profit of 0
+        if (prices == null || prices.length == 0) {
+            return 0;
+        }
+
+        // Set up initial buy and sell days and initial accumulated profit
+        int accumulatedProfit = 0;
+        int buyValue = prices[0];
+        int sellValue = prices[0];
+
+        for (int i = 0; i < prices.length; i++) {
+
+            // The day's price is lower than the best price to sell so far
+            if (prices[i] < sellValue) {
+
+                // Make the sale and add this to the profit
+                accumulatedProfit += sellValue - buyValue;
+                buyValue = prices[i];
+                sellValue = prices[i];
+            }
```

Comment:
keep your if and else if in the same bracket for readability`if(){}elseif(){}`

Reply:
got it thanks!